compiler/src/iree/compiler/API2/Embed.h

```diff
@@ -0,0 +1,257 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// Top-level compiler embedding API. This API is stable and intended to provide
+// ABI stability across releases. It should be sufficient to construct tools
+// and JITs which operate at the level of granularity of input and output
+// artifacts.
+//
+// See the bottom of the file for unstable entrypoints which allow conversion
+// to/from corresponding entities in the MLIR C API. These can be used for
+// tools that hard-link to the compiler but are not available via runtime
+// loaded stubs.
+
+#ifndef IREE_COMPILER_API2_EMBED_H
+#define IREE_COMPILER_API2_EMBED_H
+
+#include <stddef.h>
+
+#if (defined(_WIN32) || defined(__CYGWIN__))
+// Visibility annotations disabled.
+#define IREE_EMBED_EXPORTED
+#elif defined(_WIN32) || defined(__CYGWIN__)
+// Windows visibility declarations.
+#if IREE_EMBED_BUILDING_LIBRARY
+#define IREE_EMBED_EXPORTED __declspec(dllexport)
+#else
+#define IREE_EMBED_EXPORTED __declspec(dllimport)
+#endif
+#else
+// Non-windows: use visibility attributes.
+#define IREE_EMBED_EXPORTED __attribute__((visibility("default")))
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// Opaque structures.
+struct iree_compiler_error_t;
```

Comment:
you can remove the struct from all the functions below by typedefing these (`typedef struct iree_compiler_error_t iree_compiler_error_t;`) - makes the functions more readable as there's one fewer freestanding word in all the args